gpm_api/tests/dataset/generate_test_dataset_assets.py

```diff
@@ -0,0 +1,280 @@
+                subset_chunks.append(chunk)
+            else:
+                subset_size = min(subset_size, src_size)
+                subset_chunk = min(chunk, subset_size)
+                subset_shape.append(subset_size)
+                subset_chunks.append(subset_chunk)
+
+        # Determine subset shape
+        subset_shape = tuple(subset_shape)
+        subset_chunks = tuple(subset_chunks)
+    else:
+        subset_shape = h5_obj.shape
+        subset_chunks = h5_obj.chunks
+    return subset_shape, subset_chunks
+
+
+def _copy_attrs(src_h5_obj, dst_h5_obj):
+    """Copy attributes from the source file to the destination file."""
+    for key, value in src_h5_obj.attrs.items():
+        dst_h5_obj.attrs[key] = value
+
+
+def _copy_datasets(src_group, dst_group, subset_size=5):
+    for name, h5_obj in src_group.items():
+        if isinstance(h5_obj, h5py.Dataset):
+            # Determine the subset shape (2 indices per dimension)
+            subset_shape, subset_chunks = _get_subset_shape_chunks(h5_obj, subset_size=subset_size)
+
+            # Create a new dataset in the subset group with the subset shape
+            subset_dataset = dst_group.create_dataset(
+                name, subset_shape, dtype=h5_obj.dtype, chunks=subset_chunks
+            )
+
+            # Copy data from the src_h5_obj dataset to the subset dataset
+            subset_dataset[:] = h5_obj[tuple(slice(0, size) for size in subset_shape)]
+
+            # Copy attributes from the src_h5_obj dataset to the subset dataset
+            _copy_attrs(h5_obj, subset_dataset)
+
+            # Copy encoding information
+            if h5_obj.compression is not None and "compression" in h5_obj.compression:
+                subset_dataset.compression = h5_obj.compression
+                subset_dataset.compression_opts = h5_obj.compression_opts
+
+        elif isinstance(h5_obj, h5py.Group):
+            # If the h5_obj is a group, create a corresponding group in the subset file and copy its datasets recursively
+            subgroup = dst_group.create_group(name)
+            # Copy group attributes
+            _copy_attrs(h5_obj, subgroup)
+            _copy_datasets(h5_obj, subgroup, subset_size=subset_size)
+
+
+def create_test_hdf5(src_fpath, dst_fpath):
+    # Open source HDF5 file
+    src_file = h5py.File(src_fpath, "r")
+
+    # Create empty HDF5 file
+    dst_file = h5py.File(dst_fpath, "w")
+
+    # Write a subset of the source HDF5 groups and leafs into the new HDF5 file
+    _copy_datasets(src_file, dst_file, subset_size=10)
+
+    # Write attributes from the source HDF5 root group to the new HDF5 file root group
+    _copy_attrs(src_file, dst_file)
+
+    # Close connection
+    src_file.close()
+    dst_file.close()
+
+
+def cut_raw_assets():
+    filenames = os.listdir(raw_assets_dir_path)
+
+    for filename in filenames:
+        print(f"Cutting {filename}")
+        raw_asset_filepath = os.path.join(raw_assets_dir_path, filename)
```

Comment:
We should remove the raw granules and just keep the cutted ones in the GitHub repo !
Let's discuss on monday on the structure of the test_dataset data directory

Reply:
Yes that is what I thought!